rbs/CommentsAssociator.cc

```diff
@@ -86,191 +227,388 @@ void CommentsAssociator::walkNodes(parser::Node *node) {
+            auto beginLine = core::Loc::pos2Detail(ctx.file.data(ctx), node->loc.beginPos()).line;
+            auto endLine = core::Loc::pos2Detail(ctx.file.data(ctx), node->loc.endPos()).line;
+            consumeCommentsBetweenLines(beginLine, endLine);
+        },
+        [&](parser::Until *until) {
+            associateAssertionCommentsToNode(node);
+            walkNodes(until->cond.get());
+            walkNodes(until->body.get());
+            auto beginLine = core::Loc::pos2Detail(ctx.file.data(ctx), node->loc.beginPos()).line;
+            auto endLine = core::Loc::pos2Detail(ctx.file.data(ctx), node->loc.endPos()).line;
+            consumeCommentsBetweenLines(beginLine, endLine, "until");
+        },
+        [&](parser::UntilPost *untilPost) {
+            walkNodes(untilPost->cond.get());
+            walkNodes(untilPost->body.get());
+            auto beginLine = core::Loc::pos2Detail(ctx.file.data(ctx), node->loc.beginPos()).line;
+            auto endLine = core::Loc::pos2Detail(ctx.file.data(ctx), node->loc.endPos()).line;
+            consumeCommentsBetweenLines(beginLine, endLine, "until");
         },
         [&](parser::When *when) {
             walkNodes(when->body.get());
+            auto beginLine = core::Loc::pos2Detail(ctx.file.data(ctx), node->loc.beginPos()).line;
+            auto endLine = core::Loc::pos2Detail(ctx.file.data(ctx), node->loc.endPos()).line;
+            consumeCommentsBetweenLines(beginLine, endLine, "when");
+        },
+        [&](parser::While *while_) {
+            associateAssertionCommentsToNode(node);
+            walkNodes(while_->cond.get());
+            walkNodes(while_->body.get());
+            auto beginLine = core::Loc::pos2Detail(ctx.file.data(ctx), node->loc.beginPos()).line;
+            auto endLine = core::Loc::pos2Detail(ctx.file.data(ctx), node->loc.endPos()).line;
+            consumeCommentsBetweenLines(beginLine, endLine, "while");
+        },
+        [&](parser::WhilePost *whilePost) {
+            walkNodes(whilePost->cond.get());
+            walkNodes(whilePost->body.get());
+            auto beginLine = core::Loc::pos2Detail(ctx.file.data(ctx), node->loc.beginPos()).line;
             auto endLine = core::Loc::pos2Detail(ctx.file.data(ctx), node->loc.endPos()).line;
-            consumeCommentsUntilLine(endLine);
+            consumeCommentsBetweenLines(beginLine, endLine, "while");
         },
         [&](parser::Node *other) {
-            // Do nothing
+            associateAssertionCommentsToNode(node);
+            auto beginLine = core::Loc::pos2Detail(ctx.file.data(ctx), node->loc.beginPos()).line;
+            auto endLine = core::Loc::pos2Detail(ctx.file.data(ctx), node->loc.endPos()).line;
+            consumeCommentsBetweenLines(beginLine, endLine);
         });
 }
 
 std::map<parser::Node *, std::vector<CommentNode>> CommentsAssociator::run(unique_ptr<parser::Node> &node) {
+    // Remove any comments that don't start with RBS prefixes
```

Comment:
Gives around a second of speed up. I tried doing it as we iterate while consuming but kept running into seg faults.